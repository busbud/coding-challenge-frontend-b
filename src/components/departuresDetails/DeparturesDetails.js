import React from 'react';
import PropTypes from 'prop-types';
import Paper from '@material-ui/core/Paper';
import Button from '@material-ui/core/Button';
import moment from 'moment';
import './DeparturesDetails.scss';

const DeparturesDetails = props => {
    const displayTime = (time) => {
        return moment(time).format('hh:mm')
    }

    return (
        <div className="DepartureDetails">
            <Paper className="container">
                <div className="departure">
                    <div>{displayTime(props.departure.departure_time)}</div>
                    <div>{props.originLocation.name}</div>
                </div>
                <div className="bus">ðŸšŒ</div>
                <div className="destination">
                    <div>{displayTime(props.departure.arrival_time)}</div>
                    <div>{props.destinationLocation.name}</div>
                </div>
                <div className="price">
                    <div className="passenger">1 passenger</div>

                    <Button variant="contained">
                        {props.departure.prices.total}
                    </Button>
                </div>

            </Paper>
        </div>
    )
}

DeparturesDetails.propTypes = {
    departure: PropTypes.shape({
        departure_time: PropTypes.string,
        arrival_time: PropTypes.string,
        prices: PropTypes.shape({
            total: PropTypes.number
        }),
    }).isRequired,
    originLocation: PropTypes.shape({
        name: PropTypes.string
    }).isRequired,
    destinationLocation: PropTypes.shape({
        name: PropTypes.string
    }).isRequired
}

export default DeparturesDetails;