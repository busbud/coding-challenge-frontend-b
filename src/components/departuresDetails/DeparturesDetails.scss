.DepartureDetails {
    margin: 20px;
    min-width: 300px;

    .container {
        display: flex;
        flex-direction: column;
        padding: 10px;

        .departure, 
        .destination {
            display: flex;
            & :first-child {
                margin-right: 10px
            }
        }

        .departure {
            .time {
                color: #207CCB;
                font-weight: 600;
                margin-right: 20px;
            }
            .city {
                margin-right: 5px;
                font-weight: 600
            }
        }

        .destination {
            color: #B1B3B5;
            .time {
                color: #B1B3B5;
                font-weight: 600;
                margin-right: 20px;
            }
            .city {
                margin-right: 5px;
                font-weight: 600
            }
        }

        .bus {
            margin: 5px;
        }

        .price {
            display: flex;
            justify-content: space-between;
            align-items: center;
            margin-top: 10px;

            .passenger {
                color: #F19433
            }

            .total {
                color: #207CCB;
                font-size: 24px;
                align-self: flex-end;
                margin-right: 15px;
            }
        }
    } 
}