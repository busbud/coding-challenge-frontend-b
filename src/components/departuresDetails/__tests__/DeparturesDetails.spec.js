import React from 'react';
import DeparturesDetails from '../DeparturesDetails';
import renderer from 'react-test-renderer';

jest.mock('@material-ui/core/Paper', () => 'Paper');
jest.mock('@material-ui/core/Paper', () => 'Button');

const props = {
    departure: {
        departure_time: '2019-08-02T09:55:45',
        arrival_time: '2019-08-02T19:55:00',
        prices: {
            total: 25
        }
    },
    originLocation: {
        name: 'new york'
    },
    destinationLocation: {
        name: 'MontrÃ©al'
    }
}

describe('DeparturesDetails', () => {
    it('renders correctly', () => {
        const tree = renderer
            .create(<DeparturesDetails {...props} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
})
