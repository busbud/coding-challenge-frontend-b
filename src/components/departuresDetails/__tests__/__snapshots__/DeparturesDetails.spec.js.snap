// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeparturesDetails renders correctly 1`] = `
<div
  className="DepartureDetails"
>
  <Button
    className="container"
  >
    <div
      className="departure"
    >
      <div
        className="time"
      >
        09:55 AM
      </div>
      <div
        className="city"
      >
        new york
         - 
      </div>
      <div
        className="origin-location"
      >
        new york
      </div>
    </div>
    <div
      className="bus"
    >
      ðŸšŒ
    </div>
    <div
      className="destination"
    >
      <div
        className="time"
      >
        07:55 PM
      </div>
      <div
        className="city"
      >
        montrÃ©al
         -
      </div>
      <div
        className="destination-location"
      >
        MontrÃ©al
      </div>
    </div>
    <div
      className="price"
    >
      <div
        className="passenger"
      >
        1 passenger
      </div>
      <div
        className="total"
      >
         
        25
          
      </div>
    </div>
  </Button>
</div>
`;
